incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -18,18 +20,28 @@ func GetSelectorAnnotation(inst *unstructured.Unstructured) string {
 	return annot[api.AnnotationSelector]
 }
 
+func GetTreeSelectorAnnotation(inst *unstructured.Unstructured) string {
+	annot := inst.GetAnnotations()
+	return annot[api.AnnotationTreeSelector]
+}
+
 // GetTreeSelector is similar to a regular selector, except that it adds the LabelTreeDepthSuffix to every string
 // To transform a tree selector into a regular label selector, we follow these steps:
 // 1. get the treeSelector annotation if it exists
 // 2. convert the annotation string to a slice of strings seperated by comma, because user is allowed to put multiple selectors
 // 3. append the LabelTreeDepthSuffix to each of the treeSelector string
 // 4. combine them into a single string connected by comma
 func GetTreeSelector(inst *unstructured.Unstructured) (labels.Selector, error) {
-	annot := inst.GetAnnotations()
-	treeSelectorStr, ok := annot[api.AnnotationTreeSelector]
-	if !ok {
+	treeSelectorStr := GetTreeSelectorAnnotation(inst)
+	if treeSelectorStr == "" {
 		return nil, nil
 	}
+	// check if it's a valid namespace name
+	errStrs := validation.IsDNS1123Label(treeSelectorStr)
```

Comment:
`a,b` is a valid tree selector string but is not a valid DNS label

Reply:
Done. The e2e test also caught this